### PR TITLE
fix: remove process listeners after stopping the server

### PR DESCRIPTION
**Pull request by snitin315**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [x] This is a **metadata update**

### For Bugs and Features; did you add new tests?
No

### Motivation / Use-Case

fix memory leak warnings in e2e tests.

Screenshot 2021-11-07 at 5 41 57 PM

### Breaking Changes
None

### Additional Info
No

### CODE REVIEW

File: setupTest.js [hunk 1]
```diff
@@ -3,3 +3,12 @@
...
 jest.setTimeout(140000);
+
+if (expect.getState().testPath.includes("test/e2e")) {
+  afterEach(() => {
+    // We add listeners to signals when creating a new Server instance, but have no cleanup method
+    // So ensure they are removed after each run to prevent EventEmitter memory leak warnings
+    process.removeAllListeners("SIGINT");
+    process.removeAllListeners("SIGTERM");
+  });
+}
```
Comment (alexander-akait) [inline setupTest.js:14]: hm, it is because we attach to `process`? Maybe we should do it in server in close

Comment (snitin315) [reply setupTest.js:14]: Yes, we should move it to the `stop` method.

Comment (alexander-akait) [reply setupTest.js:14]: Let's do it

File: lib/Server.js [hunk 2]
```diff
@@ -2258,6 +2258,11 @@ class Server {
         this.middleware = null;
       }
     }
+
+    // We add listeners to signals when creating a new Server instance
+    // So ensure they are removed to prevent EventEmitter memory leak warnings
+    process.removeAllListeners("SIGINT");
+    process.removeAllListeners("SIGTERM");
```
Comment (alexander-akait) [inline lib/Server.js:2265]: We should not remove all of them just only our, move our close function above the file and use `removeListener`

Comment (snitin315) [reply lib/Server.js:2265]: above where?

Comment (alexander-akait) [reply lib/Server.js:2265]: I.e. just move it in function above the Server class and remove this listener in close

**Pull request merged**